pkgs/development/tools/micronaut/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchzip, jdk, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "micronaut-${version}";
+  version = "1.2.1";
+  src = fetchzip {
```

Comment:
For fetching sources `fetchurl` is preferred. Please also insert a blank line before `src` to increase readability.

Reply:
Ok this will take a little more time for me. Do you know why it is preferred? I have a few other packages I could fix for this issue.

Reply:
I actually overlooked that the URL points to Github. In that case `fetchFromGitHub` would best function to fetch the sources.

Reply:
Does fetchFromGitHub get the zip file? I really just need to get the zip and install it. Is fetchzip really not allowed here? It seems like the best solution to getting a zip file and unzip it. This is what I have done in other packages like visualvm and jbake.

Reply:
OK, I think I understand now. The zip contains the ready made jar file. I guess in that particular case it is OK to use `fetchzip`.